### PR TITLE
Fixes #2260: Added support for testing from Pypi/GitHub source archives

### LINKED ISSUE #2260: Include tests on pypi tarball
I'm updating the OpenBSD port for pywbem.  Our ports infrastructure usually get the distfiles for python ports from pypi.  It would be very useful to have the tests included on this distribution so we could execute them before packaging.

Thank you.

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message for details.

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -213,6 +230,103 @@ active, as long as the Python `tox` package is installed in it:
...
+--------------------------------------------------
+
+The wheel distribution archives on Pypi
+(e.g. ``pywbem-1.0.0-py2.py3-none-any.whl``)
+contain only the files needed to run pywbem, but not the files needed to test
+it.
+
+The source distribution archives on Pypi and GitHub
+(e.g. ``pywbem-1.0.0.tar.gz``)
+contain all files that are needed to test pywbem.
```
Comment (KSchopmeyer) [inline docs/development.rst:245]: Should this be "needed to run and to test pywbem"  or "the files to test pywbem in addition to the files to run it"

File: docs/development.rst [hunk 2]
```diff
@@ -213,6 +230,103 @@ active, as long as the Python `tox` package is installed in it:
...
+
+When installing these source distribution archives, only the files needed for
```
Comment (KSchopmeyer) [inline docs/development.rst:247]: remove the word "only"

File: docs/development.rst [hunk 3]
```diff
@@ -213,6 +230,103 @@ active, as long as the Python `tox` package is installed in it:
...
+    . . .
+    Options for 'test' command:
+      --pytest-options  additional options for pytest, as one argument
+    . . .
+
+Note: The ``test`` command of ``setup.py`` is not the deprecated built-in
+command (see `<https://github.com/pypa/setuptools/issues/1684>`_), but has been
+implemented in ``setup.py`` in such a way that it only runs the tests but
+does not install anything upfront. Similarly, the ``leaktest`` and
+``end2endtest`` commands have been implemented that way. Therefore, this
```
Comment (KSchopmeyer) [inline docs/development.rst:325]: change "that way" to "in the same way".  Remove Similarly at the beginning of that sentence

File: MANIFEST.in [hunk 4]
```diff
@@ -4,9 +4,12 @@ include README.rst
...
 include requirements.txt
+include test-requirements.txt
 include setup.py
 include build_moftab.py
 include mof_compiler
 include mof_compiler.bat
 include pywbem/*.py
 include pywbem_mock/*.py
+recursive-include tests *
+recursive-exclude tests *.pyc
```
Comment (KSchopmeyer) [inline MANIFEST.in:15]: Curosity.  Why is there a .pyc file in MANIFEST.in???

File: docs/changes.rst [hunk 5]
```diff
@@ -40,6 +40,13 @@ Released: not yet
 
 **Enhancements:**
 
+* Test: Added support for testing from Pypi/GitHub source distribution archives.
```
Comment (KSchopmeyer) [inline docs/changes.rst:43]: Suggestion: very short why this is important here would help the reader. i.e. One sentence

File: docs/development.rst [hunk 6]
```diff
@@ -213,6 +230,103 @@ active, as long as the Python `tox` package is installed in it:
...
+
+Pywbem, its dependent packages, and packages needed for testing pywbem can be
+installed with the package extra named "test":
+
+.. code-block:: bash
+
+    $ pip install .[test]
+
+When testing pywbem installations in Linux distributions that include pywbem as
+an OS-level package, these packages would instead be installed by means of
```
Comment (KSchopmeyer) [inline docs/development.rst:286]: By "these packages" I assume you mean pywbem.   The phrase "by means of OS-level packages"  do you mean "using OS-level packages" or "as an OS-Level package"?

File: docs/development.rst [hunk 7]
```diff
@@ -213,6 +230,103 @@ active, as long as the Python `tox` package is installed in it:
...
+    drwxr-xr-x  29 maiera  staff    928 Jul  3 07:55 pywbem
+    drwxr-xr-x   8 maiera  staff    256 Jul  3 07:55 pywbem.egg-info
+    drwxr-xr-x  18 maiera  staff    576 Jul  3 07:55 pywbem_mock
+    -rw-r--r--   1 maiera  staff   1067 Jun 29 22:31 requirements.txt
+    -rw-r--r--   1 maiera  staff     38 Jul  3 07:55 setup.cfg
+    -rwxr-xr-x   1 maiera  staff   7555 Jul  3 07:24 setup.py
+    -rw-r--r--   1 maiera  staff   2337 Jul  2 23:20 test-requirements.txt
+    drwxr-xr-x  15 maiera  staff    480 Jul  3 07:55 tests
+
+Pywbem, its dependent packages, and packages needed for testing pywbem can be
```
Comment (KSchopmeyer) [inline docs/development.rst:278]: Maybe I am missing it, but isn't the issue to provide for executing the pywbem tests as part of the installation whereas the PYPI installation does not do that.  I could not find that goal statement  itself in the documentation.

File: setup.py [hunk 8]
```diff
@@ -68,12 +70,101 @@ def read_file(a_file):
...
 
 
+class PytestCommand(setuptools.Command):
+    """
+    Base class for setup.py commands for executing tests for this package
+    using pytest.
+
+    Note on the class name: Because distutils.dist._show_help() shows the class
+    name for the setup.py command name instead of invoking get_command_name(),
+    the classes that get registered as commands need to have the command name.
```
Comment (KSchopmeyer) [inline setup.py:80]: Getting picky.  Change "need to have" to "must"

**Comment by andy-maier** [comment]
You did not set "Changes Requested" or the "update needed" label, but I assumed this is your complete review, and have updated the PR with resolutions as stated.

File: docs/changes.rst (hunk 5 above)
Comment (andy-maier) [reply docs/changes.rst:43]: DONE - Added this sentence:
```
  This allows testing without having to check out the entire repository, and
  is convenient for testing e.g. when packaging pywbem into OS-level packages.
```

File: docs/development.rst (hunk 6 above)
Comment (andy-maier) [reply docs/development.rst:286]: DONE - Changed the paragraph to:
```
When testing pywbem installations in Linux distributions that include pywbem as
an OS-level package, the corresponding OS-level packages would instead be
installed for these dependent Python packages. The ``test-requirements.txt``
file shows which dependent Python packages are needed for testing pywbem.
```

File: docs/development.rst (hunk 7 above)
Comment (andy-maier) [reply docs/development.rst:278]: Added the following a few lines up, as a goal statement and motivation:
```
This allows testing pywbem without having to check out the entire repository,
and is convenient for testing e.g. when packaging pywbem into OS-level packages.
```

File: setup.py (hunk 8 above)
Comment (andy-maier) [reply setup.py:80]: DONE

File: docs/development.rst (hunk 1 above)
Comment (andy-maier) [reply docs/development.rst:245]: DONE

File: docs/development.rst (hunk 2 above)
Comment (andy-maier) [reply docs/development.rst:247]: DONE

File: docs/development.rst (hunk 3 above)
Comment (andy-maier) [reply docs/development.rst:325]: DONE (both)

File: MANIFEST.in (hunk 4 above)
Comment (andy-maier) [reply MANIFEST.in:15]: NOISSUE - it is excluded, not included.

**Pull request merged**